homeassistant/components/anova/__init__.py

```diff
@@ -35,36 +34,22 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
         return False
     assert api.jwt
-    api.existing_devices = [
-        AnovaPrecisionCooker(
-            aiohttp_client.async_get_clientsession(hass),
-            device[0],
-            device[1],
-            api.jwt,
-        )
-        for device in entry.data[CONF_DEVICES]
-    ]
     try:
-        new_devices = await api.get_devices()
+        await api.create_websocket()
     except NoDevicesFound:
-        # get_devices raises an exception if no devices are online
-        new_devices = []
-    devices = api.existing_devices
-    if new_devices:
-        hass.config_entries.async_update_entry(
-            entry,
-            data={
-                **entry.data,
-                **{CONF_DEVICES: serialize_device_list(devices)},
-            },
+        _LOGGER.warning(
+            "No devices were found on the websocket, perhaps you don't have any devices on this account?"
         )
-    coordinators = [AnovaCoordinator(hass, device) for device in devices]
-    for coordinator in coordinators:
-        await coordinator.async_config_entry_first_refresh()
-        firmware_version = coordinator.data.sensor.firmware_version
-        coordinator.async_setup(str(firmware_version))
+    assert api.websocket_handler
```

Comment:
We should avoid `assert` in production code as they can be disabled by optimizations.
Is there a possibility that `websocket_handler` is None?

Reply:
There is not a possibility it is None at this point, but I have changed it to be an if statement instead.